### PR TITLE
Signals added to state machine

### PR DESCRIPTION
**Pull request by jedStevens**
I added signal emissions for when the AnimaiotnNodeStateMachinePlayback changes state. I'm not sure if I've really done this properly (in terms of style) so please feel free to suggest changes; but it does work as intended. 

Solves #22137

### CODE REVIEW

**Comment by Sslaxx** [comment]
Do all the `state_changed` signals refer to the same kind of change? Some of them sound like they could do with a more descriptive name.

**Comment by guilhermefelipecgs** [comment]
This pr can fix this #26408 as well.

File: scene/animation/animation_node_state_machine.cpp [hunk 1]
```diff
@@ -963,6 +982,10 @@ void AnimationNodeStateMachine::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_graph_offset"), &AnimationNodeStateMachine::get_graph_offset);
 
 	ClassDB::bind_method(D_METHOD("_tree_changed"), &AnimationNodeStateMachine::_tree_changed);
+
+	ClassDB::bind_method(D_METHOD("on_playback_state_changed"), &AnimationNodeStateMachine::on_playback_state_changed);
```
Comment (akien-mga) [inline scene/animation/animation_node_state_machine.cpp]: This should be prefixed with `_`.

**Comment by akien-mga** [comment]
Sorry for the late review. Commits should be squashed together and there's one change to make to the bindings. Otherwise code look fine.

**Comment by Xrayez** [comment]
Would be good if the signal provides *from* and *to* states too (either in this or future PR).
```gdscript
func _on_state_changed(from_state, to_state):
    if from_state == 'idle' and to_state == 'jump':
        leap()
```

**Comment by jedStevens** [comment]
I had some serious problems trying to squash things, I'm really unsure how to do it, so I'm going to refork and apply the changes in one commit. I'm not sure what I did to screw it up so hard...

**Pull request closed without merging**